chain/client/tests/catching_up.rs

```diff
@@ -641,4 +642,209 @@ mod tests {
+                                    let mut chunk_heights = chunk_heights.write().unwrap();
+                                    let chunk_hashes = chunk_hashes.read().unwrap();
+                                    if let Some(value) = chunk_hashes.get(&request.chunk_hash) {
+                                        chunk_heights.insert(*value);
+                                    }
+                                    return (NetworkResponses::NoResponse, false);
+                                }
+                            }
+                            if let NetworkRequests::PartialEncodedChunkMessage {
+                                account_id, ..
+                            } = msg
+                            {
+                                if account_id == "test1.2" {
+                                    return (NetworkResponses::NoResponse, false);
+                                }
+                            }
+                            if let NetworkRequests::Block { block } = msg {
+                                for chunk in block.chunks.iter() {
+                                    add_hash(chunk.hash.clone(), block.header.inner_lite.height);
+                                }
+                                println!(
+                                    "HOLDING PHASE, HEIGHT = {:?}",
+                                    block.header.inner_lite.height
+                                );
+
+                                if block.header.inner_lite.height >= height_release {
+                                    let chunk_heights = chunk_heights.read().unwrap();
+                                    let block_heights = block_heights.read().unwrap();
+                                    for i in 0..height_release + 1 {
+                                        println!("BLOCK HEIGHT FOUND = {:?}, CHUNK HEIGHT REQUESTED = {:?}", block_heights.get(&i), chunk_heights.get(&i));
+                                    }
+                                    let mut total_requests = 0;
+                                    for i in 2..height_release - 1 {
+                                        if let Some(_) = block_heights.get(&i) {
+                                            if chunk_heights.get(&i).is_some() {
+                                                total_requests += 1;
+                                            }
+                                        }
+                                    }
+                                    // This is the most important assert which checks
+                                    // that we request chunks for orphans.
+                                    // total_requests should be greater than 1 because
+                                    // requesting chunks for height 2 is trivial
+                                    // while for at least another one is not.
+                                    assert!(total_requests > 1);
```

Comment:
Should we actually estimate how many heights we expect?
What is the complexity with having this assert expect a specific value?

`> 1` doesn't seem sufficiently reliable.

Reply:
I'm not sure that this value is deterministic and we can calculate it exactly. I can set it to what's happening now. It will be a workaround, not a solution.